hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -330,8 +330,7 @@ public Pair<SchemaProvider, Pair<String, JavaRDD<HoodieRecord>>> readFromSource(
       if (lastCommit.isPresent()) {
         HoodieCommitMetadata commitMetadata = HoodieCommitMetadata
             .fromBytes(commitTimelineOpt.get().getInstantDetails(lastCommit.get()).get(), HoodieCommitMetadata.class);
-        if (cfg.checkpoint != null && (StringUtils.isNullOrEmpty(commitMetadata.getMetadata(CHECKPOINT_RESET_KEY))
-                || !cfg.checkpoint.equals(commitMetadata.getMetadata(CHECKPOINT_RESET_KEY)))) {
```

Comment:
I don't understand what you are trying to achieve with this PR. During the 3.0 -> 3.2 upgrade of App we purposely reset the checkpoint, seems like you are removing the code to do so in hudi.
A) this will break the upgrade because the checkpoint won't exists anymore in Kafka and we need to reset it
B) if you really want to remove this behavior (and you have a valid reason for it) then you should remvoe the logic in hsfs not in hudi.

Reply:
if `CHECKPOINT_RESET_KEY` == `checkpoint` then it would use the last offset, instead of the provided checkpoint.

more explanation can also be found in the Jira comments: https://hopsworks.atlassian.net/browse/FSTORE-881